# Josh_Portfolio
Data science portfolio

# [Project 1: Diabetes Prediction: Project Overview](https://github.com/JoshTjan/Diabetes_Prediction)
* In this project, we aim to predict the onset of diabetes based on diagnostic measures. We use a dataset that includes several medical predictor variables and one target variable, Outcome. Predictor variables include the number of pregnancies the patient has had, their BMI, insulin level, age, and more.
* Data Exploration: We started by exploring the data using Python. We analyzed the distributions of the features and the relationships between them. This step helped us understand the data better and informed our subsequent steps.
* Data Preprocessing: We preprocessed the data by handling missing values and scaling the features. This step is crucial as it can significantly impact the performance of the machine learning models.
* Model Training and Evaluation: We trained and evaluated several machine learning models, including Logistic Regression, Random Forest, and Support Vector Machine. We calculated several metrics, such as accuracy, precision, recall, and F1 score, to evaluate the performance of the models.
* Hyperparameter Tuning: We improved the performance of the Random Forest model by tuning its hyperparameters using grid search. Hyperparameter tuning is an important step as it can help us find the most optimal parameters for our model, thus improving its performance.

# [Project 2: Covid_Virus Genome Analysis: Project Overview](https://github.com/JoshTjan/Corona_Virus_Genome)
* The project involved the analysis of seven virus genomes, including the SARS-CoV-2 genome and six others.
* The genomes were provided in FASTA format and included AY274119.3, AY278488.2, KP876546.1, MG772933.1, MN908947.3, MN988668.1, and MN988669.1.
* The GC content of each genome was calculated, revealing relatively similar values across the genomes.
* The amino acid composition of each genome was calculated, showing that Leucine (L) was the most abundant amino acid in all genomes.
* The sequence similarity between each pair of genomes was calculated. The genomes MN988668.1 and MN988669.1 were found to be identical.
* Two motif finding tasks were performed. The 'ATG' motif was found to be common in these genomes. The 'LLQY' and 'GTNGTKR' motifs were present in some genomes but not all.
* The project provided valuable insights into the structure and composition of the seven virus genomes, revealing similarities and differences that could be further explored in future studies.
