# Josh_Portfolio
Bioinformatics portfolio

# [Project 1: Breast Cancer T-Cells Single Cell RNA Analysis: Project Overview](https://github.com/JoshTjan/Breast_Cancer_Tcells_SCRNA_Analysis_Project)
* This project focuses on the analysis of publicly available single-cell RNA sequencing (scRNA-seq) data related to breast cancer T cells. 
* The scRNA-seq data used in this project was obtained from publicly available sources. Please refer to the data source or repository for detailed information on the dataset.
* Data Preprocessing: Prior to analysis, the raw scRNA-seq data underwent a series of preprocessing steps. This included quality control, filtering out low-quality cells, and removing unwanted sources of variation.
* Normalization: To ensure that the data was comparable across all cells, normalization was performed. This step is crucial for accurate downstream analysis.
* Clustering: Clustering analysis was carried out to group cells with similar gene expression profiles. This allowed for the identification of distinct cell populations, including T cells, within the breast cancer dataset.
* Gene Identification: Genes associated with T cells were identified through differential gene expression analysis. This step aimed to highlight genes that are specifically expressed in T cells or show significant differences in expression compared to other cell types.

# [Project 2: Diabetes Prediction: Project Overview](https://github.com/JoshTjan/Diabetes_Prediction)
* In this project, we aim to predict the onset of diabetes based on diagnostic measures. We use a dataset that includes several medical predictor variables and one target variable, Outcome. Predictor variables include the number of pregnancies the patient has had, their BMI, insulin level, age, and more.
* Data Exploration: We started by exploring the data using Python. We analyzed the distributions of the features and the relationships between them. This step helped us understand the data better and informed our subsequent steps.
* Data Preprocessing: We preprocessed the data by handling missing values and scaling the features. This step is crucial as it can significantly impact the performance of the machine learning models.
* Model Training and Evaluation: We trained and evaluated several machine learning models, including Logistic Regression, Random Forest, and Support Vector Machine. We calculated several metrics, such as accuracy, precision, recall, and F1 score, to evaluate the performance of the models.
* Hyperparameter Tuning: We improved the performance of the Random Forest model by tuning its hyperparameters using grid search. Hyperparameter tuning is an important step as it can help us find the most optimal parameters for our model, thus improving its performance.


* Two motif finding tasks were performed. The 'ATG' motif was found to be common in these genomes. The 'LLQY' and 'GTNGTKR' motifs were present in some genomes but not all.
* The project provided valuable insights into the structure and composition of the seven virus genomes, revealing similarities and differences that could be further explored in future studies.
